rdtools/filtering.py

```diff
@@ -102,25 +185,325 @@ def clip_filter(power_ac, quantile=0.98):
+                        (clipping_df['subgroup'] > 1), 'value'] = True
+    elif roll_periods > 3:
+        clipping_df.loc[(~clipping_df['value']) &
+                        (clipping_df['subgroup_count'] <= 3) &
+                        (clipping_df['subgroup'] > 1), 'value'] = True
+    elif roll_periods > 1:
+        clipping_df.loc[(~clipping_df['value']) &
+                        (clipping_df['subgroup_count'] <= 2) &
+                        (clipping_df['subgroup'] > 1), 'value'] = True
+    else:
+        pass
+    clipping = clipping_df['value']
+    # High frequency was resampled to 15-minute average data.
+    # The following lines apply the 15-minute clipping filter to the
+    # original 15-minute data resulting in a clipping filter on the original
+    # data.
+    if (original_time_series_sampling_frequency < 10):
+        power_ac = power_copy.copy()
+        clipping = clipping.reindex(index=power_ac.index,
+                                    method='ffill')
+        # Subset the series where clipping filter == True
+        clip_pwr = power_ac[clipping]
+        clip_pwr = clip_pwr.reindex(index=power_ac.index,
+                                    fill_value=np.nan)
+        # Set any values within the clipping max + clipping min threshold
+        # as clipping. This is done specifically for capturing the noise
+        # for high frequency data sets.
+        daily_mean = clip_pwr.resample('D').mean()
+        daily_std = clip_pwr.resample('D').std()
+        daily_clipping_max = daily_mean + 2 * daily_std
+        daily_clipping_max = daily_clipping_max.reindex(index=power_ac.index,
+                                                        method='ffill')
+        daily_clipping_min = daily_mean - 2 * daily_std
+        daily_clipping_min = daily_clipping_min.reindex(index=power_ac.index,
+                                                        method='ffill')
+    else:
+        # Find the maximum and minimum power level where clipping is
+        # detected each day.
+        clip_pwr = power_ac[clipping]
+        clip_pwr = clip_pwr.reindex(index=power_ac.index,
+                                    fill_value=np.nan)
+        daily_clipping_max = clip_pwr.resample('D').max()
+        daily_clipping_min = clip_pwr.resample('D').min()
+        daily_clipping_min = daily_clipping_min.reindex(index=power_ac.index,
+                                                        method='ffill')
+        daily_clipping_max = daily_clipping_max.reindex(index=power_ac.index,
+                                                        method='ffill')
+    # Set all values to clipping that are between the maximum and minimum
+    # power levels where clipping was found on a daily basis.
+    final_clip = (daily_clipping_min <= power_ac) & \
+        (power_ac <= daily_clipping_max)
+    final_clip = final_clip.reindex(index=power_copy.index,
+                                    fill_value=False)
```

Comment:
I wonder if the final reindex will get us into trouble with irregular series as input. Would it work to reindex `daily_clipping_min` and `daily_clipping_max` in the conditionals above to `power_copy.index` and then here doing:
```
    final_clip = (daily_clipping_min <= power_copy) & \
        (power_copy <= daily_clipping_max)
```
work?